sales_team_multicompany/sales_team.py

```diff
@@ -0,0 +1,16 @@
+# -*- coding: utf-8 -*-
+# © 2016 Chafique DELLI @ Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, fields
+
+
+class CrmTeam(models.Model):
+    _inherit = 'crm.team'
+
+    company_ids = fields.Many2many(
+        'res.company',
+        string='Companies',
+        required=True,
```

Comment:
Don't make it required. The standard interpretation is empty for allowing all companies